clickhouse/client.cpp

```diff
@@ -1085,6 +1179,41 @@ void Client::Impl::RetryGuard(std::function<void()> func) {
     }
 }
 
+Client::PreparedInsert * Client::Impl::PrepareInsert(Query query) {
+    // Arrange a query callback to extract a block that corresponds to the
+    // query columns.
+    auto block = new Block();
+    query.OnData([&block](const Block& b) {
+        for (Block::Iterator bi(b); bi.IsValid(); bi.Next()) {
+            // Create the ClickHouse column type.
+            clickhouse::ColumnRef col = bi.Column();
+            auto chtype = col->Type();
+            if (chtype->GetCode() == Type::LowCardinality) {
+                chtype = col->As<ColumnLowCardinality>()->GetNestedType();
+            }
```

Comment:
I'm honestly not sure this is the right thing to do. Might one need Type::LowCardonality?